replace qx.lang.String.startsWith & qx.lang.String.endsWith by their prototype counterparts

schnepe2 opened the pull request:
- replaced `qx.lang.String.startsWith(foo,bar)` by `foo.startsWith(bar)`
- replaced `qx.lang.String.endsWith(foo,bar)` by `foo.endsWith(bar)`

Except for `qx.module.util.String` which I don't know what to do with.

schnepe2 committed:
- replaced qx.lang.String.startsWith(foo,bar) by foo.startsWith(bar)
- replaced qx.lang.String.endsWith(foo,bar) by foo.endsWith(bar)

level420 commented:
regarding https://github.com/qooxdoo/qooxdoo/blob/master/framework/source/class/qx/module/util/String.js#L95

I'd wrap it via a function body like this:

```javascript
    /**
     * Check whether the string ends with the given substring.
     *
     * @attachStatic {qxWeb, string.endsWith}
     * @signature function(fullstr, substr)
     * @param fullstr {String} the string to search in
     * @param substr {String} the substring to look for
     * @return {Boolean} whether the string ends with the given substring
     */
    endsWith : function(fullstr, substr) {
        return fullstr.endsWith(substr);
    },
```
as this preserves API compatibility.

schnepe2 committed:
qx.module.util.String replacements

wrapped 'qx.module.util.String.startsWith' and 'qx.module.util.String.endsWith'  in a function body

level420 commented:
Looks good! 
Start voting.

level420 commented:
5 thumbs up.
merging
Thank you @schnepe2
